types/service/database_service_configuration.go

```diff
@@ -372,6 +437,93 @@ func (config GcpCloudSqlConnectorConfiguration) Validate() error {
 	return fmt.Errorf("invalid database authentication type: %s", config.AuthenticationType)
 }
 
+// SQLServerDatabaseServiceConfiguration represents service configuration for Microsoft SQL Server databases.
+//
+// Border0 currently supports two ways of connecting to Microsoft SQL Server databases.
+// Use the corresponding configuration fields to configure the upstream connection.
+type SQLServerDatabaseServiceConfiguration struct {
+	HostnameAndPort
+
+	Kerberos          *UsernameAndPassword `json:"kerberos_configuration,omitempty"`
+	SqlAuthentication *UsernameAndPassword `json:"sql_authentication_configuration,omitempty"`
+}
+
+// Validate ensures that the `AzureSqlDatabaseServiceConfiguration` is valid.
+func (config SQLServerDatabaseServiceConfiguration) Validate() error {
+	if err := config.HostnameAndPort.Validate(); err != nil {
+		return err
+	}
+
+	switch {
+	case config.Kerberos != nil:
+		if nilcheck.AnyNotNil(config.SqlAuthentication) {
+			return errors.New("authentication type is kerberos_configuration, but sql_authentication_configuration is provided")
+		}
+
+		return nil
+	case config.SqlAuthentication != nil:
+		if nilcheck.AnyNotNil(config.Kerberos) {
+			return errors.New("authentication type is sql_authentication_configuration, but kerberos_configuration is provided")
+		}
+
+		return nil
+	default:
+		return errors.New("one of the following authentication types is required: kerberos, sql_authentication")
+	}
+}
+
+// AzureSqlDatabaseServiceConfiguration represents service configuration for Azure SQL Server databases.
+//
+// Border0 currently supports four ways of connecting to Azure SQL Server databases.
+// Use the corresponding configuration fields to configure the upstream connection.
+type AzureSqlDatabaseServiceConfiguration struct {
+	HostnameAndPort
+
+	AzureActiveDirectoryPassword   *UsernameAndPassword `json:"azure_active_directory_password_configuration,omitempty"`
+	AzureActiveDirectoryIntegrated *struct{}            `json:"azure_active_directory_integrated_configuration,omitempty"`
```

Comment:
can we make this a boolean?

Reply:
I think we will add fields to the struct later, like region or file locations (simular to aws config)

Reply:
In that case i think for now it makes more sense to just remove the field if its not used for anything.